supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesTask.scala

```diff
@@ -226,21 +239,26 @@ final case class TesTask(jobDescriptor: BackendJobDescriptor,
 }
 
 object TesTask {
+  // Helper to determine which source to use for a workflowExecutionIdentity
+  def resourcesFromWorkflowExecutionIdentity(runtimeAttributes: TesRuntimeAttributes,
+                                             configIdentity: Option[String],
+                                             runtimeIdentity: Option[String]): Resources = {
+    (configIdentity, runtimeIdentity) match {
+      case (Some(configId), _) => TesTask.makeResources(runtimeAttributes, Some(configId))
+      case (None, runtimeId) => TesTask.makeResources(runtimeAttributes, runtimeId)
```

Comment:
I really like having this logic separated from `makeResources`, but it might be clearer if it lived in a separate place by itself. You could have a third `val` that checked the two workflow execution identities and returned the proper one, and then pass that directly into `makeResources`.